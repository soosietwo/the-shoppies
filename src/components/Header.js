import React from "react";
import { TopBar } from "@shopify/polaris";
import SearchBar from "./SearchBar";

import { gql, useQuery } from "@apollo/client";

export const NOMINEES_CONNECTION_QUERY = gql`
  query NOMINEES_CONNECTION_QUERY {
    nomineesConnection {
      aggregate {
        count
      }
    }
  }
`;

const Header = ({ toggleSheetActive, nomineesCount }) => {
  const { loading, error, data } = useQuery(NOMINEES_CONNECTION_QUERY);
  const count = data ? data.nomineesConnection.aggregate.count : 0;

  return (
    <TopBar
      showNavigationToggle
      searchField={<SearchBar />}
      secondaryMenu={
        <TopBar.Menu
          activatorContent={
            <span>
              Nominees
              <span
                style={{
                  padding: "5px 9px",
                  borderRadius: "50%",
                  background: "red",
                  color: "white",
                  marginLeft: "5px",
                  fontSize: "1.2rem",
                }}
              >
                {count}
              </span>
            </span>
          }
          onOpen={toggleSheetActive}
          onClose={toggleSheetActive}
        />
      }
    />
  );
};

export default Header;
