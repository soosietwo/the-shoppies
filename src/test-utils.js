// test-utils.js
import React from "react";
import { render } from "@testing-library/react";
import { AppProvider, Frame } from "@shopify/polaris";
import en from "@shopify/polaris/locales/en.json";
import Store from "./store";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const AllTheProviders = ({ children }) => {
  return (
    <AppProvider i18n={en}>
      <Store>
        <Frame>{children}</Frame>
      </Store>
    </AppProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
