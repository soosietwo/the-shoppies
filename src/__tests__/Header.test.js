import React from "react";
import { render, waitFor } from "../test-utils";
import { MockedProvider } from "@apollo/client/testing";
import Header, { NOMINEES_CONNECTION_QUERY } from "../components/Header";

const mocks = [
  {
    request: { query: NOMINEES_CONNECTION_QUERY },
    result: {
      data: {
        nomineesConnection: {
          aggregate: {
            count: 1,
          },
        },
      },
    },
  },
];

describe("Header", () => {
  it("renders and matches snapshot", () => {
    const { container } = render(
      <MockedProvider addTypename={false} mocks={mocks}>
        <Header />
      </MockedProvider>
    );

    expect(container.querySelector(".Polaris-TopBar")).toMatchSnapshot();
  });

  it("displays the correct number of nominees", async () => {
    const { container } = render(
      <MockedProvider addTypename={false} mocks={mocks}>
        <Header />
      </MockedProvider>
    );

    await waitFor(() => {
      expect(container.querySelector(".count")).toHaveTextContent("1");
    });
  });
});
