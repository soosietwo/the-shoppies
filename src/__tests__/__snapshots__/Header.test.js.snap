// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders and matches snapshot 1`] = `
<div
  class="Polaris-TopBar"
>
  <button
    aria-label="Toggle menu"
    class="Polaris-TopBar__NavigationIcon"
    type="button"
  >
    <span
      class="Polaris-Icon"
    >
      <svg
        aria-hidden="true"
        class="Polaris-Icon__Svg"
        focusable="false"
        viewBox="0 0 20 20"
      >
        <path
          d="M19 11H1a1 1 0 0 1 0-2h18a1 1 0 1 1 0 2zm0-7H1a1 1 0 0 1 0-2h18a1 1 0 1 1 0 2zm0 14H1a1 1 0 0 1 0-2h18a1 1 0 0 1 0 2z"
        />
      </svg>
    </span>
  </button>
  <div
    class="Polaris-TopBar__Contents"
  >
    <div
      class="Polaris-TopBar__SearchField"
    >
      <div
        class="Polaris-TopBar-SearchField"
      >
        <span
          class="Polaris-VisuallyHidden"
        >
          <label
            for="PolarisSearchField1"
          >
            Search
          </label>
        </span>
        <input
          autocapitalize="off"
          autocomplete="off"
          autocorrect="off"
          class="Polaris-TopBar-SearchField__Input"
          id="PolarisSearchField1"
          placeholder="Search"
          type="search"
          value=""
        />
        <span
          class="Polaris-TopBar-SearchField__Icon"
        >
          <span
            class="Polaris-Icon"
          >
            <svg
              aria-hidden="true"
              class="Polaris-Icon__Svg"
              focusable="false"
              viewBox="0 0 20 20"
            >
              <path
                d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8zm9.707 4.293l-4.82-4.82A5.968 5.968 0 0 0 14 8 6 6 0 0 0 2 8a6 6 0 0 0 6 6 5.968 5.968 0 0 0 3.473-1.113l4.82 4.82a.997.997 0 0 0 1.414 0 .999.999 0 0 0 0-1.414z"
              />
            </svg>
          </span>
        </span>
        <div
          class="Polaris-TopBar-SearchField__Backdrop Polaris-TopBar-SearchField__BackdropShowFocusBorder"
        />
      </div>
    </div>
    <div
      class="Polaris-TopBar__SecondaryMenu"
    >
      <div>
        <div
          class="Polaris-TopBar-Menu__ActivatorWrapper"
        >
          <button
            aria-controls="Polarispopover1"
            aria-expanded="false"
            aria-owns="Polarispopover1"
            class="Polaris-TopBar-Menu__Activator"
            tabindex="0"
            type="button"
          >
            <div
              style="display: flex; align-items: center;"
            >
              Nominees
              <span
                class="sc-gtssRu hdlOKH"
              >
                <div>
                  <div
                    class="sc-bdnylx czgcBA count"
                  >
                    0
                  </div>
                </div>
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
